keymanagementscenarios.md

```diff
@@ -0,0 +1,154 @@
+    - Container pulled from any registry is be validated with listed root public keys before execution.
+- Specify location of trusted public root keys
+    - Deployer Admin gets location of key repository from publisher.
+    - Deployer Admin adds location of key repository to runtime configuration.
+    - Container pulled from any registry is be validated with listed root public keys before execution.
+- Enterprises can restrict users to add/remove their root public keys to the trust store used across a fleet of runtime hosts. The trust store will validate images pulled from any registry.
+- Enterprises can restrict users to add/remove root public keys for trusted third parties to their trust stores. The trust store will validate images pulled from any registry.
+- Air gapped environments can either specify the 
+
+Key Distribution Workflows:
+- Manual Configuration
+    - Publisher places their root certificate in a public location. This could be a website similar to what public CAs today provide.
+    - Developer copies certificate from public location and adds it to their trust store. Developer can configure whether they trust this certificate for all their artifacts, individual registries, individual repositories, or individual targets.
+- Automated Configuration
+    - Publisher gets intermediate certificate from a key distriution service i.e public CA or registry operator.
+    - Developer adds root certificate and location of key respoitory to their trust store. Developer can configure whether they trust certificates for all their artifacts, individual registries, individual repositories, or individual targets.
+- Revocation: Note in the event a root certificate is revoked validations will fail, and developers will need to update their root certificate. The publisher will need to pro-actively communicate a rotation out of bands to prevent an outage.    
+
+
+Key rotation/revocation use cases:
+
+Key Revocation Workflow
+
+![Notaryv2 Revocation Workflow](./media/nv2-keymgmt-architecture-revokekeys-v1.png)
+
+- Root Revocation (compromised root should not be needed in process to designate itself as revoked, otherwise attacker can use compromised root for a key rotation locking out publisher)
+    - Publisher deletes revoked root public key on registry (registry can stop sharing containers with revoked key).
+    - [TODO: Discuss if we want this] Registry stops vending containers signed with old root key (will prevent revoked artifacts from being used by developers not checking signature).
+    - Publisher removes revoked root public key from shared list (other users stop trusting artifacts with revoked key, provides defense in depth as they are not relying on registry).
+    - Publisher creates new root key following any scenario listed earlier in Signing use cases (starting from scratch as old key can no longer be trusted).
+    - Publisher uploads new root public key to registry (registry can verify user for container uploads). 
```

Comment:
So… what actually authenticates the root revocation? “The registry” (which one?) accepting the user’s login? A HTTPS-protected web site read by humans saying to replace the root key? Intentionally not specified by this document?

> compromised root should not be needed in process to designate itself as revoked, otherwise attacker can use compromised root for a key rotation locking out publisher

I’m confused by this: if access to the original root is not necessary, what authenticates the revocation? At a first glance this makes things easier for the attacker but not much for the publisher, unless I guess the publisher lost the key.

Reply:
Authentication of the root is deliberately done through a separate channel. Using a HTTPS site is the recommended option. In the event of a root compromise/rotation, the new root needs to verified through the same channel. The attacker will need to compromise both the root communication channel as well as the root key. Essentially a root key rotation will force some manual action on the part of the deployer so they can verify if the rotation is legitimate.